### DISCUSSION TITLE
How to apply +noise to apply it as transparency?

### DISCUSSION THREAD

**Discussion by Zeioth** [ImageMagick/ImageMagick#5917] [Help]
### ImageMagick version

7.1.0-55

### Operating system, version and so on

arch

### Description

I want to apply the +noise as transparency (as a mean to reduce the amount of ink used when printing). 

What would be the right syntax for this?

**Comment by fmw42** [comment]
What kind of noise?  Random, Gaussian, etc?

Assume Gaussian.  We read the input. Then copy it and cover it with white (-fill white -colorize 100). Then we attenuate by 1 (no change, larger than one emphasizes the noise, smaller hides the noise). Then we extract just the green channel for a grayscale image. We then put the grayscale image into the alpha channel of the input using -compose copy_opacity -composite. And finally save the result.

`magick image.png \( +clone -fill white -colorize 100 -attenuate 1 +noise gaussian -channel g -separate \) -alpha off -compose copy_opacity -composite result.png
`

Change the fill color gray or black depending upon what you want. Change the attenuate value to increase or decrease the noise.
